test/index.cpp

```diff
@@ -1590,3 +1590,8 @@ TEST(Index, Issue1867ChainedIndexingLeak)
                       &lock_bytes, &lock_buffers);
     ASSERT_EQ(0u, lock_buffers);
 }
+
+TEST(Index, Issue1452)
+{
+    EXPECT_THROW(af::seq(15,-15), af::exception);
```

Comment:
I believe `af::seq(15, -15)` is a valid sequence. Isn't it just `af::seq(15, af::end-14)` ?

If I am not missing something (and that is a big if! :-D ), than only the signs can be tested. E.g.,

```c++
    af::seq( 1,  1,  1);
    af::seq( 1,  5,  1);
    af::seq( 1, -5,  1);
    af::seq(-1,  5, -1);
    af::seq(-1, -5, -1);

    EXPECT_THROW(af::seq( 1,  1, -1), af::exception);
    EXPECT_THROW(af::seq( 1,  5, -1), af::exception);
    EXPECT_THROW(af::seq( 1, -5, -1), af::exception);
    EXPECT_THROW(af::seq(-1,  5,  1), af::exception);
    EXPECT_THROW(af::seq(-1, -5,  1), af::exception);
```